packages/headless/src/utils/expression-builder.ts

```diff
@@ -0,0 +1,30 @@
+function createExpressionBuilder(expression = '') {
+  const parts = expression ? [expression] : [];
+
+  return {
+    addStringFieldExpression(
+      field: string,
+      operator: '==' | '!=',
+      values: string[]
+    ) {
+      const stringValues = values.map((v) => `"${v}"`).join(',');
+      const expression = `${field}==(${stringValues})`;
+      const prefix = operator === '!=' ? 'NOT ' : '';
+
+      parts.push(`${prefix}${expression}`);
+    },
+
+    join(delimiter: 'AND' | 'OR') {
+      const concatenated = parts.join(`) ${delimiter} (`);
+      return `(${concatenated})`;
+    },
+  };
+}
+
+const builder = createExpressionBuilder();
+builder.addStringFieldExpression('filetype', '==', ['pdf']);
+builder.addStringFieldExpression('author', '!=', ['alice', 'bob']);
+const expression = builder.join('AND');
+
+console.log(expression);
+// (filetype==("pdf")) AND (NOT author==("alice","bob"))
```

Comment:
Here is the [v4 api](https://z22n2.csb.app/#operation/search) if you want to compare with the new syntax. It seems quite complex to build something with, because it involves composing any of 12 different object types.